platforms/org.eclipse.mita.platform.xdk110/src/org/eclipse/mita/platform/xdk110/connectivity/UserStaticSizeInferrer.xtend

```diff
@@ -0,0 +1,36 @@
+package org.eclipse.mita.platform.xdk110.connectivity
+
+import org.eclipse.emf.ecore.EObject
+import org.eclipse.mita.program.GeneratedFunctionDefinition
+import org.eclipse.mita.program.SignalInstance
+import org.eclipse.mita.program.inferrer.ElementSizeInferrer
+import org.eclipse.mita.program.inferrer.StaticValueInferrer
+import org.eclipse.mita.program.model.ModelUtils
+import org.yakindu.base.expressions.expressions.FeatureCall
+
+class UserStaticSizeInferrer extends ElementSizeInferrer {
+	
+	override infer(EObject obj) {
```

Comment:
This one is provided as an example, but I agree that since it's very general it could be useful in lots of places. However I'm not sure what the loader does when the class to be loaded is not in the same plugin as the platform component, there is a comment in there that made me wary.

Reply:
After trying it out it seems plugins can load classes not in themselves, so sharing this inferrer is possible. However I wouldn't recommend to make it a default inferrer the main ESI inherits from since this could create unexpected behaviour and is not very much in the spirit of plug-in generators/validators/inferrers.
I propose instead to make this ESI available in `org.eclipse.mita.program.inferrer` where platforms can load it from.

Reply:
I agree, making this the parent of the main ESI might yield unexpected behaviour. I was proposing to 
* make it the default ESI for components that do not specify one, and to 
* make it the recommendation for users wanting to implement their own ESI.

The latter of the two I am no longer sure is a good idea.